src/haproxy.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+#
+# igcollect - haproxy
+#
+# Copyright (c) 2017 InnoGames GmbH
+#
+
+from argparse import ArgumentParser
+from subprocess import Popen, PIPE
+from time import time
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('--prefix', default='haproxy')
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    haproxy_info = Popen(
+        ('echo "show stat" | socat /var/run/haproxy.stat stdio'),
```

Comment:
It is not cool to run shell commands for things we can do in Python.  You can use `stdin=PIPE`, and `proc.communicate()`.

Reply:
Maybe we can do this `socat` on Python as well to avoid calling any shell command.

Reply:
https://stackoverflow.com/questions/17226912/why-does-simple-echo-in-subprocess-not-working